lib/sles4sap_publiccloud_basetest.pm

```diff
@@ -25,28 +26,19 @@ sub cleanup {
     return if ($self->{cleanup_called});
     $self->{cleanup_called} = 1;
 
-    for my $command ('ansible', 'terraform') {
-        # Skip cleanup if ansible inventory is not present (deployment could not have been done without it)
-        next if (script_run 'test -f ' . qesap_get_inventory(get_required_var('PUBLIC_CLOUD_PROVIDER')));
+    qesap_upload_logs();
+    if ($self->{network_peering_present}) {
+        delete_network_peering();
+    }
+    my @cmd_list;
+    # Skip the Ansible cleanup if ansible inventory is not present (deployment could not have been done without it)
+    push(@cmd_list, 'ansible') if (script_run 'test -f ' . qesap_get_inventory(get_required_var('PUBLIC_CLOUD_PROVIDER')));
 
-        if (check_var('IS_MAINTENANCE', 1)) {
-            record_info('Cleanup', `Executing peering cleanup (if peering is present)`);
-            if (is_azure) {
-                my $rg = qesap_az_get_resource_group();
-                my $ibsm_rg = get_required_var('IBSM_RG');
-                # Check that required vars are available befor delleting the peering
-                if (defined $rg && $rg ne '' && defined $ibsm_rg && $ibsm_rg ne '') {
-                    qesap_az_vnet_peering_delete(source_group => $rg, target_group => $ibsm_rg);
-                }
-                else {
-                    record_info('No peering', 'No peering exists, peering destruction skipped');
-                }
-            }
-            elsif (is_ec2) {
-                my $deployment_name = qesap_calculate_deployment_name();
-                qesap_aws_delete_transit_gateway_vpc_attachment(name => $deployment_name . '*');
-            }
-        }
+    # Terraform destroy can be executed in any case
+    push(@cmd_list, 'terraform');
+
+    for my $command (@cmd_list) {
+        next if (script_run 'test -f ' . qesap_get_inventory(get_required_var('PUBLIC_CLOUD_PROVIDER')));
```

Comment:
I think the whole next line should go right? 
The inventory test is now handled in L35.